### DISCUSSION TITLE
How to install Scrutiny Monitor on Casa OS?

### DISCUSSION THREAD

**Discussion by iBGMike** [IceWhaleTech/CasaOS#1491] [Q&A] [answered]
Hello. I needed an application to monitor disks in the Casa OS, so I decided to use Scrutiny [https://github.com/AnalogJ/scrutiny](url) . I installed it through Docker Compose, it installed, but nothing is displayed. I can’t figure out what I’m doing wrong, my technical knowledge is low. Maybe someone has already installed Scrutiny and knows how to do it correctly?

If there are any other disk monitoring applications that can be installed in Casa OS, I would consider the options.
Thx.

**Comment by korobkinn** [comment, marked as answer]
Hi @iBGMike ,
I'd managed to install it through "Install a customized app" option of CasaOS WebUI, and importing docker CLI config from [repository readme](https://github.com/AnalogJ/scrutiny) you provided.

Here is brief instruction:
### 1. Go to "Install a customized add" -> "Import" (button on top right of popped-up window) -> "Docker CLI":

<img width="684" alt="Screenshot_1" src="https://github.com/IceWhaleTech/CasaOS/assets/26652157/7c40f0ba-cb36-4cb1-bd2d-11f5a8e6aa18">
<img width="399" alt="Screenshot_2" src="https://github.com/IceWhaleTech/CasaOS/assets/26652157/bbfd22cd-14bd-4c47-b986-c02f066b9799">

### 2. Modify Docker CLI command for your environment.

As an example, in my case, final command will be:
> docker run -it --rm -p **38080**:8080 -p **38086**:8086 \
>   -v **/DATA/AppData/scrutiny/config**:/opt/scrutiny/config \
>   -v **/DATA/AppData/scrutiny/influxdb2**:/opt/scrutiny/influxdb \
>   -v /run/udev:/run/udev:ro \
>   --cap-add SYS_RAWIO \
>   --device=/dev/sda \
>   --device=**/dev/nvme0** \
>   **--privileged**
>   --name scrutiny \
>   ghcr.io/analogj/scrutiny:master-omnibus

What I've changed in my case (marked **bold** in my example):

- WebUI ports: **8080->38080** and **8086->38080** to avoid probable interference with other CasaOS apps.
- Binded exact app path for CasaOS: **/DATA/AppData/scrutiny/config** and **/DATA/AppData/scrutiny/influxdb2**
- Added **/dev/nvme0** device for monitoring also my NVME SSD
- Added **--privileged** mode to access NVME S.M.A.R.T. (in my cases without this mode NVME drive did not appear in the UI)

### 3. Insert Docker CLI command into the window, press "Submit", and "Ok" in appeared pop-up:
<img width="328" alt="Screenshot_6" src="https://github.com/IceWhaleTech/CasaOS/assets/26652157/ecfee0af-2a76-4dcf-b810-b95616aa92eb">
<img width="285" alt="Screenshot_7" src="https://github.com/IceWhaleTech/CasaOS/assets/26652157/b4383cc8-e01b-499e-bae1-613b291f6aa0">

### 4. Check imported settings and add required options.

In my case I had to:
- Add app title (required)
- Add app icon (optional), I used [link ](https://raw.githubusercontent.com/AnalogJ/scrutiny/master/webapp/frontend/src/assets/images/logo/scrutiny-logo-dark.png)from repo.
- Choose WebUI port
- Fix first volume host path (in my case some why after import it was /DATA/AppData//opt/scrutiny/config except /DATA/AppData/scrutiny/config)
- Manually add **SYS_RAWIO** capability (it was not imported, you must start to type it in the field, and select `SYS_RAWIO`, that it will appear in the field as a tag)
See my screen:
<img width="202" alt="Screenshot_8" src="https://github.com/IceWhaleTech/CasaOS/assets/26652157/f7287ebf-14b7-461b-a7b5-7f2ebe86d77a">

### 5. (Hopefully) Enjoy!
<img width="677" alt="Screenshot_9" src="https://github.com/IceWhaleTech/CasaOS/assets/26652157/187a69fb-1604-43a8-ae40-ef52cb9a1375">
<img width="727" alt="Screenshot_11" src="https://github.com/IceWhaleTech/CasaOS/assets/26652157/e68d72de-dfee-4f9d-9135-69b75023da08">

**Comment by iBGMike** [comment]
Wow! Thanks, it worked. Just now I noticed my mistake, volume host path I had /DATA/AppData**//**opt/scrutiny/config instead of /DATA/AppData/scrutiny/config I didn’t even pay attention.
